python/src/multi_agent_orchestrator/agents/parallel_agent.py

```diff
@@ -0,0 +1,131 @@
+from multi_agent_orchestrator.agents import (
+    Agent,
+    AgentOptions,
+    BedrockLLMAgent,
+)
+from multi_agent_orchestrator.types import ConversationMessage, ParticipantRole
+from multi_agent_orchestrator.utils.logger import Logger
+
+
+# Extend AgentOptions for ParallelAgent class:
+class ParallelAgentOptions(AgentOptions):
+    def __init__(
+        self,
+        agents: list[str],
+        default_output: str = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.agents = agents
+        self.default_output = default_output
+
+
+# Create a new custom agent that allows for parallel processing:
+class ParallelAgent(Agent):
+    def __init__(self, options: ParallelAgentOptions):
+        super().__init__(options)
+        self.agents = options.agents
+        self.default_output = (
+            options.default_output or "No output generated from the ParallelAgent."
+        )
+        if len(self.agents) == 0:
+            raise ValueError("ParallelAgent requires at least 1 agent to initiate!")
+
+    async def _get_llm_response(
+        self,
+        agent: BedrockLLMAgent,
+        input_text: str,
+        user_id: str,
+        session_id: str,
+        chat_history: list[ConversationMessage],
+        additional_params: dict[str, str] = None,
+    ) -> str:
+        # Get response from LLM agent:
+        final_response: ConversationMessage | AsyncIterable[Any]
+
+        try:
+            response = await agent.process_request(
+                input_text, user_id, session_id, chat_history, additional_params
+            )
+            if self.is_conversation_message(response):
+                if response.content and "text" in response.content[0]:
+                    final_response = response
+                else:
+                    Logger.warn(f"Agent {agent.name} returned no text content.")
+                    return self.create_default_response()
+            elif self.is_async_iterable(response):
+                Logger.warn("Streaming is not allowed for ParallelAgents!")
+                return self.create_default_response()
+            else:
+                Logger.warn(f"Agent {agent.name} returned an invalid response type.")
+                return self.create_default_response()
+
+        except Exception as error:
+            Logger.error(
+                f"Error processing request with agent {agent.name}: {str(error)}"
+            )
+            raise f"Error processing request with agent {agent.name}: {str(error)}"
+
+        return final_response
+
+    async def process_request(
+        self,
+        input_text: str,
+        user_id: str,
+        session_id: str,
+        chat_history: list[ConversationMessage],
+        additional_params: dict[str, str] = None,
+    ) -> ConversationMessage:
+        # Create tasks for all LLMs to run in parallel:
+        tasks = []
+        for agent in self.agents:
+            tasks.append(
+                self._get_llm_response(
```

Comment:
Why adding another method? Can't you just call agent.process_request()?

Reply:
Wanted to include some of the `ChainAgent` logic in lines 41-65 [here](https://github.com/awslabs/multi-agent-orchestrator/blob/070906803c0859940a5df50cca1901f2d9800a72/python/src/multi_agent_orchestrator/agents/chain_agent.py#L41) in the async function that gets run for each individual agent within the `ParallelAgent`, which seemed easier/cleaner in a new internal method.  If we think it's unnecessary, fine to revise or remove